File: CSharp/SubscriptionApi/SubscriptionApi/ResponseBuilders/SubscriptionStartResponseBuilder.cs
using System;
using SubscriptionApi.Dto;
using SubscriptionApi.Queries;

namespace SubscriptionApi.ResponseBuilders
{
  public interface SubscriptionStartResponseBuilder : 
    SubscriptionValidationResults, 
    QueryResolutionEvents, 
    AssetAuthorizationEvents
  {
    StartSubscriptionResponseDto Build();
    void AssertNoFatalErrors(Exception exceptionToThrow);
  }
}

File: Java/factories/_01_no_factories/src/main/java/ResponseBuilders/SubscriptionStartResponseBuilder.java
package responseBuilders;

import dto.StartSubscriptionResponseDto;
import queries.QueryResolutionEvents;

public interface SubscriptionStartResponseBuilder extends
    SubscriptionValidationResults,
    QueryResolutionEvents,
    AssetAuthorizationEvents {
    StartSubscriptionResponseDto buildStart();
    void assertNoFatalErrors(RuntimeException exceptionToThrow);
}
